velox/exec/MergeJoin.cpp

```diff
@@ -96,22 +96,41 @@ void MergeJoin::initializeFilter(
   auto numFields = filter_->expr(0)->distinctFields().size();
   names.reserve(numFields);
   types.reserve(numFields);
+
+  // When 'channel' in the 'filter_' is found in 'inputType', first try to add
+  // the projection if it's in 'output_', otherwise, add it to filter input
+  // projections. Finally, add it's name and type to 'names' and 'types'.
+  auto addChannel = [&](column_index_t channel,
+                        const std::vector<IdentityProjection>& outputs,
+                        std::vector<IdentityProjection>& filters,
+                        const RowTypePtr& inputType) {
+    bool inOutputs{false};
```

Comment:
I think we can rewrite this logic a bit to make it shorter.

```
    names.emplace_back(inputType->nameOf(channel));
    types.emplace_back(inputType->childAt(channel));
    for (const auto [inputChannel, outputChannel] : outputs) {
      if (inputChannel == channel) {
        filterInputToOutputChannel_.emplace(filterChannel++, outputChannel);
        return;
      }
    }

    filters.emplace_back(channel, filterChannel++);
```

Reply:
Nice refinement, updated.